src/components/elements/typography.ts

```diff
@@ -85,7 +100,13 @@ export const Text = styled.p`
 /////////////////// TEXT IS DEPRECATED. PLEASE USE HEADER. ///////////////////
 //////////////////////////////////////////////////////////////////////////////
 
-export const Header = styled(({ as = 'h2', children, ...props }) =>
+type HeaderProps = {
+    as?: string
```

Comment:
```suggestion
    as?: React.ElementType;
```
what do you think about this approach @NikitK-deriv ? 
this is just a suggestion, but if you add it we will get better type inference :pray:

Reply:
Unfortunately we can't, because pass a string only here. To make seo optimization better. And sharing what exatcly I get by this way. Also I tried `React.ElementType | string` but no luck with that
<img width="1236" alt="Screenshot 2022-09-06 at 12 04 57" src="https://user-images.githubusercontent.com/103182473/188598687-a622c357-3d72-47b8-866a-e38531b47583.png">